### PR TITLE
Tests: implement non-global noparallel marks 

### PR DESCRIPTION
**Pull request by rcarpa**
It is now possible to have "noparallel groups". Tests with the same
group will never be run in parallel, but two different groups can be
run on two different nodes. If no groups are provided to
@pytest.mark.noparallel, the old behavior is maintained: that test
cannot be run concurrently at all.

### CODE REVIEW

**Comment by bari12** [comment]
Please rebase

**Review by bari12 — DISMISSED** [review]

**Comment by bari12** [comment]
Looks good to me. Maybe @maany have a second look as well, since quite a few things in tests are changing.

File: pyrightconfig.json [hunk 1]
```diff
@@ -3,8 +3,7 @@
         "lib"
     ],
     "exclude": [
-        "lib/rucio/db",
-        "lib/rucio/tests",
```
Comment (maany) [inline pyrightconfig.json:7]: Do we want to run pyright on tests?

File: etc/docker/dev/README.rst [hunk 2]
```diff
@@ -54,9 +54,9 @@ To verify that everything is in order, you can now either run the full unit test
 Alternatively, you can bootstrap the test environment once with the `-i` option and then selectively or repeatedly run test case modules, test case groups, or even single test cases, for example::
 
     tools/run_tests_docker.sh -i
-    tools/pytest.sh test_replica.py
-    tools/pytest.sh -vvv test_replica.py::TestReplicaCore
-    tools/pytest.sh -vvv --full-trace test_replica.py::TestReplicaCore::test_delete_replicas_from_datasets
+    tools/pytest.sh tests/test_replica.py
```
Comment (maany) [inline etc/docker/dev/README.rst:57]: Just a slight note, for the donkeyrider, we will have to probably specify the values for `--filter-tests` in the tests/<filter> format ;) 
https://github.com/rucio/rucio/blob/81bacae53dbe2486bcff87d55feb0dc65daa7ac0/tools/test/donkeyrider.py#L126C9-L126C9 

Not sure what would be the correct place to document it. Maybe the `donkeyrider.py` itself?

File: pyrightconfig.json (hunk 1 above)
Comment (rcarpa) [reply pyrightconfig.json:7]: It only runs inside "lib". See a couple of lines above

File: etc/docker/dev/README.rst (hunk 2 above)
Comment (rcarpa) [reply etc/docker/dev/README.rst:57]: I'm not sure I understand correctly the issue. Is it just about the paths in filter-tests changing ?

**Review by maany — DISMISSED** [review]
I've reviewed the changes to the rucio pytest plugin and the new test groups. They look good to go.

**Review by bari12 — APPROVED** [review]

**Pull request merged**